balancer-js/src/modules/pools/pool-types/concerns/composableStable/exit.concern.integration.spec.ts

```diff
@@ -0,0 +1,182 @@
+  let tokensBalanceBefore: BigNumber[];
+  let tokensBalanceAfter: BigNumber[];
+  let tokensMinBalanceIncrease: BigNumber[];
+  let signerAddress: string;
+
+  // Setup chain
+  before(async function () {
+    await forkSetup(
+      signer,
+      pool.tokensList,
+      Array(pool.tokensList.length).fill(0),
+      Array(pool.tokensList.length).fill(
+        parseFixed(initialBalance, 18).toString()
+      ),
+      jsonRpcUrl as string,
+      16350000 // holds the same state as the static repository
+    );
+    signerAddress = await signer.getAddress();
+  });
+
+  const testFlow = async (
+    [to, data, maxBPTIn, minAmountsOut]: [string, string, string, string[]],
+    exitTokens: string[]
+  ) => {
+    // Check balances before transaction to confirm success
+    [bptBalanceBefore, ...tokensBalanceBefore] = await getBalances(
+      [pool.address, ...exitTokens],
+      signer,
+      signerAddress
+    );
+    // Get expected balances out of transaction
+    bptMaxBalanceDecrease = BigNumber.from(maxBPTIn);
+    tokensMinBalanceIncrease = minAmountsOut.map((a) => BigNumber.from(a));
+    // Send transaction to local fork
+    const transactionResponse = await signer.sendTransaction({
+      to,
+      data,
+      gasLimit: 3000000,
+    });
+    transactionReceipt = await transactionResponse.wait();
+
+    // Check balances after transaction to confirm success
+    [bptBalanceAfter, ...tokensBalanceAfter] = await getBalances(
+      [pool.address, ...exitTokens],
+      signer,
+      signerAddress
+    );
+  };
+
+  context('exitExactBPTIn', async () => {
+    context('single token max out', async () => {
+      before(async function () {
+        const bptIn = parseFixed('10', 18).toString();
+        const { to, data, minAmountsOut } = controller.buildExitExactBPTIn(
+          signerAddress,
+          bptIn,
+          slippage,
+          false,
+          pool.tokensList.filter((address) => address !== pool.address)[0]
+        );
+        await testFlow(
+          [to, data, bptIn, minAmountsOut],
+          pool.tokensList.filter((address) => address !== pool.address)
+        );
+      });
+
+      it('should work', async () => {
+        expect(transactionReceipt.status).to.eql(1);
+      });
+
+      it('tokens balance should increase by at least minAmountsOut', async () => {
+        for (let i = 0; i < tokensBalanceAfter.length; i++) {
+          expect(
+            tokensBalanceAfter[i]
+              .sub(tokensBalanceBefore[i])
+              .gte(tokensMinBalanceIncrease[i])
+          ).to.be.true;
+        }
+      });
+
+      it('bpt balance should decrease by exact bptMaxBalanceDecrease', async () => {
+        expect(bptBalanceBefore.sub(bptBalanceAfter).eq(bptMaxBalanceDecrease))
+          .to.be.true;
+      });
+    });
+  });
+  context('exitExactTokensOut', async () => {
+    context('single token out', async () => {
+      before(async function () {
+        amountsOut = tokensOut.map((t, i) => {
```

Comment:
This can be a const and remove earlier declaration as its only used in this scope?